### PR TITLE
Dropbox token error passes to output

### PR DESCRIPTION
**Pull request by matacoder**
Just wrap it in `try` `except` and pass it to notifier.

Resolves: #65 
Now we see output and Blackbox is working even with bad Dropbox token.
image

### CODE REVIEW

**Review by lemonsaurus — CHANGES_REQUESTED** [review]
Okay, what you're actually doing here seems fine, but there is an antipattern in this file now and I think we can do a little bit better.

As a rule of thumb, we don't want to wrap big chunks of code in `try/except` statements. We want to wrap _as little code as possible_ in the try.

For example, in `rotate()`, only `self.client.files_list_folder` and `self.client.files_delete` can actually fail with a `BadInputError`. In `sync`, things are a little bit more spread out, but the same principle applies.

Now, if we were to wrap each line that might fail because of a bad token with a separate `try/except`, we'd end up with a whole bunch of `try/except` statements littered all over this file, which is not great. So, I have another suggestion.

Maybe in the `__init__`, we validate that we actually have a valid token? If we don't, there's no need to do anything at all when `sync` is called. Creating tempfiles and compressing stuff and whatever else we're doing is just a waste of time at that point, it's gonna fail anyway.

so basically: 
- When Dropbox object is instantiated, validate that the token works. Don't do this inside the `__init__` method itself, write a helper method for it and call that from `__init__`.
- If the token doesn't work, put the error in `self.output`, set `self.success` to `False`, and just **disable the whole client.** Check if `self.disabled` is True at the beginning of `sync` and `rotate` and just return if it is.
- When it comes to uploading files, this might still hit ApiErrors and HttpErrors for networking reasons (it might be a large file), so we should probably keep the try/except in `sync` the way it is, just minus the BadInputError, which we will check during `__init__`

File: blackbox/handlers/storage/dropbox.py [hunk 1]
```diff
@@ -26,9 +27,25 @@ def __init__(self, **kwargs):
 
         self.upload_base = self.config.get("upload_directory") or "/"
         self.client = DropboxClient(self.config["access_token"])
+        self.valid = self._validate_token()
+
+    def _validate_token(self):
+        """Check if dropbox token is valid."""
+        try:
+            return self.client.check_user("test").result == "test"
```
Comment (lemonsaurus) [inline blackbox/handlers/storage/dropbox.py:35]: Sweet! Very nice that you found and used the dropbox clients built-in way of doing this! Extremely elegant solution.

File: blackbox/handlers/storage/dropbox.py [hunk 2]
```diff
@@ -26,9 +27,25 @@ def __init__(self, **kwargs):
...
+        except AuthError:
+            return False
 
     def sync(self, file_path: Path) -> None:
         """Sync a file to Dropbox."""
+        # Check if Dropbox token is valid.
+        if self.valid is False:
+            error = "Dropbox token is invalid"
```
Comment (lemonsaurus) [inline blackbox/handlers/storage/dropbox.py:43]:
```suggestion
            error = "Dropbox token is invalid!"
```

File: blackbox/handlers/storage/dropbox.py [hunk 3]
```diff
@@ -84,6 +101,10 @@ def rotate(self, database_id: str) -> None:
         are older than `retention_days`, and because of this,
         it's better to have backups in isolated folder.
         """
+        # Check if Dropbox token is valid.
+        if self.valid is False:
+            log.warning("Dropbox token is invalid. Can't delete old backups")
```
Comment (lemonsaurus) [inline blackbox/handlers/storage/dropbox.py:106]:
```suggestion
            log.error("Dropbox token is invalid - Can't delete old backups!")
```

File: blackbox/handlers/storage/dropbox.py [hunk 4]
```diff
@@ -26,9 +27,25 @@ def __init__(self, **kwargs):
...
+            return False
 
     def sync(self, file_path: Path) -> None:
         """Sync a file to Dropbox."""
+        # Check if Dropbox token is valid.
+        if self.valid is False:
+            error = "Dropbox token is invalid"
+            self.success = False
+            self.output = error
+            log.warning(error)
```
Comment (lemonsaurus) [inline blackbox/handlers/storage/dropbox.py:46]: I don't think warning is the correct log level. 
- A warning is for informing users that something unexpected happened, but everything should still be working more or less as normal.
- An error is for informing the user that something is definitely wrong, but it will not prevent the entire application from running.. probably.
- Critical is for errors that prevent the entire application from running.

Comment (lemonsaurus) [inline blackbox/handlers/storage/dropbox.py:46]:
```suggestion
            log.error(error)
```

**Review by lemonsaurus — CHANGES_REQUESTED** [review]
Just a few minor requests, but this looks great! I really like this solution now.

File: blackbox/handlers/storage/dropbox.py (hunk 4 above)
Comment (matacoder) [reply blackbox/handlers/storage/dropbox.py:46]: Thank you, will use the correct level from now on.
